File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        int bCount = 0;
        int aCount = 0;
        int lCount = 0;
        int oCount = 0;
        int nCount = 0;
        for (int i = 0; i < text.length(); i++) {
            if (text[i] == 'b') {
                bCount += 1;
            } else if (text[i] == 'a') {
                aCount += 1;
            } else if (text[i] == 'l') {
                lCount += 1;
            } else if (text[i] == 'o') {
                oCount += 1;
            } else if (text[i] == 'n') {
                nCount += 1;
            }
        }
        lCount = lCount / 2;
        oCount = oCount / 2;
        return min({bCount, aCount, lCount, oCount, nCount});
    }
};

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        min_count = float('inf')
        # ToDo: Write Your Code Here.
        count = {}
        for c in text:
            if c not in count:
                count[c] = 0
            count[c] += 1
        min_count = min(min_count, count.get('b', 0))
        min_count = min(min_count, count.get('a', 0))
        min_count = min(min_count, count.get('l', 0) // 2)
        min_count = min(min_count, count.get('o', 0) // 2)
        min_count = min(min_count, count.get('n', 0))
        
        return min_count